Gems/Maestro/Code/Source/Cinematics/AnimAZEntityNode.cpp

```diff
@@ -19,198 +19,190 @@
-        }
     }
-    else
+
+    /// @deprecated Serialization for Sequence data in Component Entity Sequences now occurs through AZ::SerializeContext and the Sequence
+    /// Component
+    void CAnimAzEntityNode::Serialize(XmlNodeRef& xmlNode, bool bLoading, bool bLoadEmptyTracks)
     {
-        // saving
-        if (m_entityId.IsValid())
+        CAnimNode::Serialize(xmlNode, bLoading, bLoadEmptyTracks);
+        if (bLoading)
         {
-            AZ::u64 id64 = static_cast<AZ::u64>(m_entityId);
-            xmlNode->setAttr("AnimatedEntityId", id64);
+            AZ::u64 id64;
+            if (xmlNode->getAttr("AnimatedEntityId", id64))
+            {
+                m_entityId = AZ::EntityId(id64);
+            }
+        }
+        else
+        {
+            // saving
+            if (m_entityId.IsValid())
+            {
+                AZ::u64 id64 = static_cast<AZ::u64>(m_entityId);
+                xmlNode->setAttr("AnimatedEntityId", id64);
+            }
         }
     }
-}
 
-//////////////////////////////////////////////////////////////////////////
-void CAnimAzEntityNode::SetSkipInterpolatedCameraNode(const bool skipNodeCameraAnimation)
-{
-    // Skip animations on transforms
-    CAnimComponentNode* transformNode = GetTransformComponentNode();
-    if (transformNode)
+    void CAnimAzEntityNode::SetSkipInterpolatedCameraNode(const bool skipNodeCameraAnimation)
     {
-        transformNode->SetSkipComponentAnimationUpdates(skipNodeCameraAnimation);
-    }
+        // Skip animations on transforms
+        CAnimComponentNode* transformNode = GetTransformComponentNode();
+        if (transformNode)
+        {
+            transformNode->SetSkipComponentAnimationUpdates(skipNodeCameraAnimation);
+        }
 
-    // Skip animations on cameras
-    CAnimComponentNode* cameraComponentNode = GetComponentNodeForComponentWithTypeId(AZ::Uuid(EditorCameraComponentTypeId));
-    if (!cameraComponentNode)
-    {
-        cameraComponentNode = GetComponentNodeForComponentWithTypeId(AZ::Uuid(CameraComponentTypeId));
+        // Skip animations on cameras
+        CAnimComponentNode* cameraComponentNode = GetComponentNodeForComponentWithTypeId(AZ::Uuid(EditorCameraComponentTypeId));
+        if (!cameraComponentNode)
+        {
+            cameraComponentNode = GetComponentNodeForComponentWithTypeId(AZ::Uuid(CameraComponentTypeId));
+        }
+        if (cameraComponentNode)
+        {
+            cameraComponentNode->SetSkipComponentAnimationUpdates(skipNodeCameraAnimation);
+        }
     }
-    if (cameraComponentNode)
+
+    void CAnimAzEntityNode::Reflect(AZ::ReflectContext* context)
     {
-        cameraComponentNode->SetSkipComponentAnimationUpdates(skipNodeCameraAnimation);
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<CAnimAzEntityNode, CAnimNode>()->Version(1)->Field("Entity", &CAnimAzEntityNode::m_entityId);
```

Comment:
The previous style is actually the modern side when we use the constructor/builder pattern like this.